changelogs/unreleased/compiler-bugfix-list-comprehension-unknowns.yml

```diff
@@ -0,0 +1,7 @@
+description: "Compiler: fixed bug where a list comprehension over an Unknown (``[x for x in <unknown>]``) would break the compiler under specific circumstances"
+change-type: patch
+sections:
+  bugfix: "{{description}}"
```

Comment:
I think we need to indicate the behavior change, so that in case it goes wrong, it is apparent form the changelog.

Reply:
Apparent is a bit of a stretch. It's still very abstract. But I'll give it a shot.